### PR TITLE
Materials restructuring

### PR DESCRIPTION
**Pull request by nerdinand**
This is a big one. Some highlights:
* Sessions are now called lessons (this distinguishes them from our actual meetup "sessions")
* Lessons now have the following things:
  * a description
  * slides
  * exercises
  * additional_materials (e.g. cheat sheets or legacy slides and exercises)
* Exercises are now web pages instead of PDFs, that will make them way easier to edit
* The old (new) curriculum and topic pages are gone (they have morphed into the new structure)
* Exercises now have hints at the top (general hint, IRB hint)
* Lesson 1 and 2 have had their exercises converted

Here's what the lesson overview looks like now for session 1 and 2:
Screenshot 2020-02-04 at 17 35 49

Here's an example exercise:
Screenshot 2020-02-04 at 17 38 08
Screenshot 2020-02-04 at 17 38 15

(Maybe somebody with some design talent can make it look prettier eventually)

### CODE REVIEW

File: _includes/lesson.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    <small>
+      {{lesson.topic}}
+    </small>
+  </h2>
+
+  <p>{{lesson.description}}</p>
+
+  <ul>
+    {% for slide in lesson.slides %}
+      <li><a href="http://rubymonstas.ch/slides.rubymonstas.ch/slides/{{slide[0]}}/">Slides: {{ slide[1] }}</a> (<a href="https://github.com/rubymonstas-zurich/slides.rubymonstas.ch/releases/latest/download/{{slide[0]}}.pdf">PDF</a>)</li>
```
Comment (crackofdusk) [inline _includes/lesson.html]: Do I understand correctly that with the following yaml:

```yaml
slides:
     welcome: Welcome slides
```
`slide[0]` returns `welcome` and `slide[1]` `Welcome slides`?

File: _includes/lesson.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    {% endfor %}
+  </ul>
+
+  <ul>
+    {% for exercise in lesson.exercises %}
+      <li><a href="/exercises/{{ lesson.number }}/{{ exercise[0] }}">Exercises: {{ exercise[1] }}</a></li>
```
Comment (crackofdusk) [inline _includes/lesson.html]: What do you think of using headings for slides, exercises and additional materials? (I know it requires more conditionals)

File: exercises/2/recap-1.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+layout: exercises
+topic: Recap lesson 1
+---
+
+{% include exercises/irb_hint.html %}
+
+1. You have two strings: ​`"Ruby is"`​ and `​"great!"​`. Combine them to one string which says: ​`"Ruby is great!"`. Figure out how to concatenate strings using the official documentation or using Google.
+
+2. Strings and Numbers are two different things. The string ​`"42"​` and the number ​`42​` are not the same.
```
Comment (crackofdusk) [inline exercises/2/recap-1.md]:
```suggestion
2. Strings and numbers are two different things. The string ​`"42"​` and the number ​`42​` are not the same.
```

File: _includes/lesson.html (hunk 1 above)
Comment (nerdinand) [reply _includes/lesson.html]: yes, exactly... liquid is a weird template language...

File: _includes/lesson.html (hunk 2 above)
Comment (nerdinand) [reply _includes/lesson.html]: yeah, i thought about it, but i think it's just going to be too many headings on this page. it's not really obvious what belongs to what anymore... maybe we can experiment a bit more and find something that works...

File: exercises/1/methods.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+layout: exercises
+topic: Methods
+---
+
+{% include exercises/irb_hint.html %}
+
+1. In Ruby, methods names can end with a `?`. What's special about these kinds of methods?
+2. Look at the official [Ruby documentation for String](https://ruby-doc.org/core/String.html). Find 3 methods that end with `?`. Read the documentation about them to see what they do and try them out in IRB!
+3. Let's try chaining some methods. Take the following String: `"Good work, well done!"`. There is a String method which allows you to remove runs of the same character. E.g. `"Good"` becomes `"God"`. Figure out this method's name using the String documentation. Call this method on the string, then figure out if the resulting new string has an even number of characters using method chaining. In the end you'll have joined 3 methods after the string.
```
[the remark this answers is not in the document]
Comment (nerdinand) [reply exercises/1/methods.md]: The example in the slides for method chaining is literally `"I love Ruby".length.odd?`. So I think they should be able to figure it out...

File: exercises/2/comparison-operators.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+layout: exercises
+topic: Comparison Operators
+---
+
+{% include exercises/irb_hint.html %}
+
+1. Comparison operators can not only be used with numbers. What happens if you compare strings with the comparison operators `<` and `>`? What rules are applied here?
+
+2. Try to compare two strings with `==`. What are the semantics of doing that?
```
[the remark this answers is not in the document]
Comment (nerdinand) [reply exercises/2/comparison-operators.md]: How would you formulate the question? Is "What does this achieve?" better?

**Comment by nerdinand** [comment]
@hharen You really are a fan of giant PDFs, aren't you? ;P
We don't currently have a way to generate PDFs for the exercises. Though I'm sure that's possible somehow. It would just have to be automated, as we want to get away from the manual work we had in the past.
However, it could also be confusing to see that link on the website. I'm not convinced ;)
Maybe we can ask some of our students what they prefer...

File: exercises/1/methods.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (hharen) [reply exercises/1/methods.md]: yep

**Comment by nerdinand** [comment]
I also converted lesson 3's exercises now. If you can think of any more/better exercises for hashes, that would be very welcome!

**Pull request merged**